docs/DeveloperGenerateTest.md

```diff
@@ -0,0 +1,91 @@
+# Generating Tests
+
+Before generating tests you should be aware of the following environment variables that are used to render new test files from a template:
+
+| Variable | Required | Explanation | Default Value |
+|---|---|---|---|
+| `TEMPLATE_SUITE_PACKAGE_NAME` | yes | The name of the package directory and spec file created in `tests/` and the name used when creating the suite file in `cmd/`  | `template`  |
+| `TEMPLATE_SPEC_FILE_NAME` | no | It will override the name of the test spec file and the text within the `Describe/It` containers. Useful when you are looking to test a different component within a service. Refer to `tests/build/` for a good example.    | `template` |
+| `TEMPLATE_APPEND_FRAMEWORK_DESCRIBE_FILE` | no | By default, when the test spec file is generated, the root `Describe` container function is appended to the `pkg/framework/describe.go` file, using the value of either `TEMPLATE_SUITE_PACKAGE_NAME` or `TEMPLATE_SPEC_FILE_NAME`, so it can be utilized in the current e2e test suite. In some cases, due the type of testing, you may not want to perform this action. | `true`  |
+| `TEMPLATE_JOIN_SUITE_SPEC_FILE_NAMES` | no | It will join the values of `TEMPLATE_SUITE_PACKAGE_NAME` and `TEMPLATE_SPEC_FILE_NAME` to be used for the `pkg/framework/describe.go` root `Describe` container function and in all text within the Gingko container text | `false` |
+
+ 
+ ## Generating Gingko Test Spec File
+  
+ ```bash
+   $ make local/template/generate-test-spec
+   ./mage -v local:generateTestSpecFile
+   Running target: Local:GenerateTestSpecFile
+   I1128 12:49:18.543491    1661 magefile.go:337] TEMPLATE_SUITE_PACKAGE_NAME not set. Defaulting test suite package directory as template.
+   I1128 12:49:18.543542    1661 magefile.go:341] TEMPLATE_SPEC_FILE_NAME not set. Defaulting test spec file to value of as `template`.
+   I1128 12:49:18.543547    1661 magefile.go:345] TEMPLATE_APPEND_FRAMEWORK_DESCRIBE_FILE not set. Defaulting to true which will update the pkg/framework/describe.go as.
+   I1128 12:49:18.543689    1661 magefile.go:373] Creating new test package directory and spec file tests/template/template.go.
+   exec: go "fmt" "tests/template/template.go"
+   tests/template/template.go
+   exec: go "fmt" "pkg/framework/describe.go"
+ ```
+ This command will create a new package under the `tests/`directory and a test spec file `<specfile>.go` for you. It will contain some basic imports but more importantly it will generate a basic structured Ginkgo like the one below. You can update as necessary and build upon it.
+
+ ```golang
+ package chaos
```

Comment:
nit: in the example above, there's no mention about using the package name "chaos" before running the mage target. could be a bit confusing, but not a big deal.

Reply:
ack, I'll update this to be more clear